### PR TITLE
Feature/set dev environment

### PR DESCRIPTION
**Pull request by luizoamorim**
Feature to set our dev environment with tools like: eslint, prettier, editorconfig, commitzen and husky.

Settings done:
- Eslint with standard configuration;
- Editorconfig for standardize for different operational systems and dev tools;
- Prettier for code style;
- Husky to trigger lint-stage (eslint fix and prettier fix) and commitzen;
- [Commitzen](https://www.conventionalcommits.org/en/v1.0.0/): Allow us to follow tha convetional commits pattern for our commits.
- Typescript settings

### CODE REVIEW

**Review by daveroga — APPROVED** [review]
LGTM

**Review by imagobea — CHANGES_REQUESTED** [review]
Thank you @luizoamorim for all these plugins!
Have probably added enough comments to the PR. As a summary:

- Few files are missing the final new line (looks like there is a flag somewhere that might help)
- Let's discuss development practices and update the README

Many thanks again! :tada:

File: package.json [hunk 1]
```diff
@@ -1,10 +1,24 @@
 {
   "name": "nightfall-sdk",
-  "version": "beta.xxxx",
+  "version": "0.1.0",
```
[the remark this answers is not in the document]
Comment (luizoamorim) [reply package.json:3]: Yes. Generated an error with eslint.

File: README.md [hunk 2]
```diff
@@ -3,3 +3,33 @@
...
+
+We are following the convetional commits specification for our commits.
+https://www.conventionalcommits.org/en/v1.0.0/
+For this we're using commitzen to help us in the commits creation.
+
+### Branches pattern
+
+We're following the github flow standard. So we have the master branch that is our releasy ready branch and for features and bugfixes we use the following:
+
+TBD...
```
[the remark this answers is not in the document]
Comment (luizoamorim) [reply README.md:35]: I agree with you. Let's put things like that in some internal quick guide.

File: package.json [hunk 3]
```diff
@@ -21,5 +35,27 @@
     "reconnecting-websocket": "^4.4.0",
     "web3": "^1.7.3",
     "ws": "^8.6.0"
+  },
+  "devDependencies": {
```
[the remark this answers is not in the document]
Comment (luizoamorim) [reply package.json:39]: Ok. Fine.

File: .editorconfig [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+# top-most EditorConfig file
+root = true
+
+[*]
+indent_style = space
+indent_size = 2
+end_of_line = lf
+charset = utf-8
+trim_trailing_whitespace = false
+insert_final_newline = false
```
[the remark this answers is not in the document]
Comment (luizoamorim) [reply .editorconfig:12]: I've changed for true.

**Pull request merged**